File: Repository/Repositories/Staff/CommunicationLists/CommunicationListEntryModel.cs
namespace Repository.Repositories.Staff;

public class CommunicationListEntryModel: CommunicationListEntryItem
{
    public string UserName { get; set; }
    public string UserEmailAddress { get; set; }
    public string GradeLevel { get; set; }

    public string GuardianName { get; set; }
    public string GuardianEmailAddress { get; set; }

    public string SecondaryGuardianName { get; set; }
    public string SecondaryGuardianEmailAddress { get; set; }

    public string MentorName { get; set; }
    public string MentorEmail { get; set; }

    public bool IsStaff { get; set; }
}

File: Web/Angular/ClientApp/src/app/models/communication-list-entry.model.ts
export interface CommunicationListEntry {
  // from item
  communicationListId?: number;
  userId: number;
  includeStudent?: boolean;
  includeGuardian1?: boolean;
  includeGuardian2?: boolean;
  includeMentor?: boolean;
  includeStaff?: boolean;
  staffId?: number;

  // from model
  userName: string;
  userEmailAddress: string;
  guardianName?: string;
  guardianEmailAddress?: string;
  secondaryGuardianName?: string;
  secondaryGuardianEmailAddress?: string;
  mentorName?: string;
  mentorEmail?: string;
  gradeLevel?: string;

  // client-side only
  included: boolean;
  includeSelected?: boolean;
  removeSelected?: boolean;
  additionalRecipientEmailAddress: string;

  isStaff: boolean;
}
